src/DynamoDb/Marshaler.php

```diff
@@ -0,0 +1,236 @@
+<?php
+namespace Aws\DynamoDb;
+
+use GuzzleHttp\Stream\StreamInterface;
+
+/**
+ * Marshals and unmarshals JSON documents and PHP arrays into DynamoDB items.
+ */
+class Marshaler
+{
+    /**  @var callable Called if the marshaler encounters an invalid value. */
+    private $safetyNetFn;
+
+    /**
+     * Instantiates the DynamoDB Marshaler.
+     *
+     * The `$safetyNetFn` allows the user to provide custom logic to handle
+     * invalid values (e.g., an empty string). The signature of this function
+     * should look like `function (&$type, &$value) : bool`. It should modify
+     * `$type` and `$value` as needed in order to marshal the value to a valid
+     * DynamoDB attribute. The function should return `true` when successful,
+     * or `false` on failure, so that the normal error state can be reached.
```

Comment:
Well, you need to be able to change the type and value. Maybe it could return the same type of array that marshalValue returns or null. That might be better.